### ISSUE TITLE
Validate format of value set using Setter or during FromJson/FromYaml

### ISSUE THREAD

**Issue by ashutshkumr** [open-traffic-generator/openapiart#156]
e.g. following should throw error

```go

e := config.Flows().Items()[0].Packet().Add().SetEthernet()
// raise error later
e.SetSrc().SetValue("1.1.1.1")

// handle the same for FromJson()
```

**Comment by ajbalogh** [comment]
Proposal: All run time validation errors be accumulated into a global private validation object.  

Two methods can be provided to provide validation feedback:
1) a top level Validate() func will return nil, or an error with all validation errors
2) if Validate() isn't called then it will be called during any rpc
The Validate() func should return a full list of all errors, not just the first one encountered. 

```go
api := NewApi()
config := api.NewConfig()
e := config.Flows().Items()[0].Packet().Add().Ethernet()

// the following should add errors to the internal global validation object 
e.Src().SetValue("1.1.1.1")
e.Dst().SetValue("00:00")

// method 1
// err will have the full list of validation errors if any
if err := Validate(); errs != nil {
    panic(err)
}

// method 2
// err will have the full list of validation errors if any, or if no validation errors then any transport/server errors
resp, err := api.SetConfig(config)
if err != nil {
    panic(err)
}
```

**Comment by ashutshkumr** [comment]
> Proposal: All run time validation errors be accumulated into a global private validation object.
> 
> Two methods can be provided to provide validation feedback:
> 
> 1. a public Validate() func will return nil, or an error with all validation errors or should it panic with all validation content.
>    `Why even allow execution to continue if the configuration is invalid? `
> 2. if Validate() isn't called then it will be called during any rpc
>    The Validate() func should return a full list of all errors, not just the first one encountered.
> 
> ```go
> api := NewApi()
> config := api.NewConfig()
> e := config.Flows().Items()[0].Packet().Add().Ethernet()
> 
> // the following should add errors to the internal global validation object 
> e.Src().SetValue("1.1.1.1")
> e.Dst().SetValue("00:00")
> 
> // method 1
> // err will have the full list of validation errors if any
> if err := Validate(); errs != nil {
>     panic(err)
> }
> 
> // method 2
> // err will have the full list of validation errors if any, or if no validation errors then any transport/server errors
> resp, err := api.SetConfig(config)
> if err != nil {
>     panic(err)
> }
> ```

Agree with the proposal, but there's third scenario missing.

We also need to internally call `Validate()` when calling `ToJson()`/`ToYaml()`/`ToPbText()`

**Comment by Rangababu-R** [comment]
available in v0.1.13
